src/plugins/main.js

```diff
@@ -105,6 +120,45 @@ export default async (context) => {
     }
   }
 
+  const loadAndSetLocale = async (newLocale, init = false) => {
```

Comment:
I really don't like this `init` argument. It's a very generic name that changes code flow in unclear (unless you dig into the code) ways.

I don't really have much better suggestion but at least we could make it a bit clearer at call site by using options object:
```suggestion
  const loadAndSetLocale = async (newLocale, { initialSetup = false } = {}) => {
```